### ISSUE TITLE
System include paths in `compile_commands.json` arguments fed to `cl.exe` are not recognized

### ISSUE THREAD

**Issue by jeremyong** [microsoft/vscode-cpptools#10169]
### Environment

- OS and Version: Windows 10 Pro 21H1
- VS Code Version: 1.73.1 (user setup)
- C/C++ Extension Version: 1.13.4 (Pre-Release)
- Other extensions you installed: Reproduces without other extensions installed
- With a minimal single `.cpp` file and `compile_commands.json` workspace, `/external:I` flags are not recognized

### Bug Summary and Steps to Reproduce

Bug Summary:

Include paths in system folders are not recognized by the intellisense engine, with or without the `/external:` prefix (e.g. `"/external:I\"C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\DIA SDK\\include\""`). In this example, with the following `compile_commands.json` file:

```json
[
    {
        "file": "PDB.cpp",
        "arguments": [
  "C:\\PROGRA~1\\MIB055~1\\2022\\COMMUN~1\\VC\\Tools\\MSVC\\14.34.31933\\bin\\Hostx64\\x64\\cl.exe",
          "/nologo",
          "/c",
          "/JMC",
          "/ZI",
          "/std:c++20",
          "/Zc:__cplusplus",
          "/permissive-",
          "/D_DEBUG",
          "/MDd",
          "/Od",
          "/external:I\"C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\DIA SDK\\include\"",
          "/Fd:D:\\Test\\PDB.obj.pdb",
          "/Fo:D:\\Test\\PDB.obj",
          "D:\\Test\\PDB.cpp"
        ],
        "directory": "D:\\Test"
    }
]
```

And the following source file:

```c++
// PDB.cpp
#include <Windows.h>

#include <dia2.h>

int main()
{

}
```

Intellisense reports that `dia2.h` cannot be located:

![image](https://user-images.githubusercontent.com/250149/202808388-8c336c96-46d4-4885-90f4-1a9cb3dde564.png)

Note that no overrides of any config are specified, and there is no c_cpp properties json or workspace settings either.

### Expected behavior

The `<dia.h>` header should be available (verified in the path specified in the include path argument above).

### Code sample and Logs

C/C++ logs reveal no errors, but identifies the translation unit as below:

```shell
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
sending compilation args for D:\Test\PDB.cpp
  include: C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\TOOLS\MSVC\14.34.31933\INCLUDE
  include: C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\TOOLS\MSVC\14.34.31933\ATLMFC\INCLUDE
  include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.22621.0\UM
  include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.22621.0\UCRT
  include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.22621.0\SHARED
  include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.22621.0\WINRT
  include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.22621.0\CPPWINRT
  define: _DEBUG
  define: _MT
  define: _DLL
  define: _DEBUG
  other: --ms_cplusplus_std_value
  other: --no_ms_permissive
  stdver: ms_c++20
  intelliSenseMode: windows-msvc-x64
Queueing IntelliSense update for files in translation unit of: D:\Test\PDB.cpp
cpptools/finishUpdateSquiggles
Error squiggle count: 3
```
```

**Comment by Colengms** [comment]
Hi @jeremyong .  I think the issue is with shell quoting in that argument, causing your `compile_commands.json` to not be well formed.  The syntax for that argument should be:

```
          "/external:IC:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\DIA SDK\\include",
```

Clarification has been added to [the specification for compile_commands.json here](https://clang.llvm.org/docs/JSONCompilationDatabase.html), indicating that the `command` field may contain shell quoting/escaping, but the  `arguments` arrays must not.  I believe that clarification was motivated by issues reported here in this repo.  (See https://github.com/microsoft/vscode-cpptools/issues/8649#issuecomment-1010314847 )

We came to a similar conclusion elsewhere in the extension, that only a command-line may contain shell escaping/quoting, and that arrays of arguments must not contain shell escaping/quoting.

I'm able to repro a similar issue, but am able to resolve it by removing the superfluous quotes.

Let us know if that doesn't resolve the issue for you.
